lib/money/money.rb

```diff
@@ -316,17 +316,7 @@ def split(num)
   # @example
   #   Money.new(100, "USD").calculate_splits(3) #=> {Money.new(34) => 1, Money.new(33) => 2}
   def calculate_splits(num)
-    raise ArgumentError, "need at least one party" if num < 1
-    subunits = self.subunits
-    low = Money.from_subunits(subunits / num, currency)
-    high = Money.from_subunits(low.subunits + 1, currency)
-
-    num_high = subunits % num
-
-    {}.tap do |result|
-      result[high] = num_high if num_high > 0
-      result[low] = num - num_high
-    end
+    Splitter.new(self, num).split.dup
```

Comment:
What do you think of the idea of being able to call either `to_h` or `to_a` on this Splitter object depending on the format you want? 
```suggestion
    Splitter.new(self, num).to_h
```
to me it's easier to reason about than `split`, which is unclear if it's an array. Long term we could even deprecate this `calculate_splits` method all together in favor of `money.split(3).to_h`

Reply:
This isn't a proper implementation for `to_h`. This object is an `Enumerable` that replaces an Array and tries to act as much like an Array as possible. 

If you define `to_h` like this you'll break lots of code and assumptions. `Splitter#to_h` should behave like `Array#to_h`.